mat_nn_openmp.hpp

```diff
@@ -233,9 +257,11 @@ double su3_mat_nn(std::vector<site> &a, std::vector<su3_matrix> &b, std::vector<
 
 #endif
 
-  ttotal = std::chrono::duration_cast<std::chrono::microseconds>(Clock::now()-tstart).count();
   } // end of OpenMP block, C gets moved back to the host
 
+  ttotal = std::chrono::duration_cast<std::chrono::microseconds>(Clock::now()-tstart).count();
+  profile->d2h_time = (std::chrono::duration_cast<std::chrono::microseconds>(Clock::now()-tprofiling).count())/1.0e6;
```

Comment:
We should add a target exit data directive above this line, and set tprofiling to clock::now before it.